### ISSUE TITLE
Toil fails to run when /var/user/ is not writable

### ISSUE THREAD

**Issue by oneillkza** [DataBiosphere/toil#4307]
Hi there,

I'm trying to run Progressive Cactus, on the latest Toil (5.7.1) on Slurm, but keep running into errors like:

```
[2023-01-03T14:48:10-0800] [MainThread] [I] [cactus.shared.common] Running the command ['singularity', '--silent', 'run', '/projects/CanSeq/scratch/cactus/2023-01-03_14:46:39/jobstore/cactus.img', 'cactus_sanitizeFastaHeaders', 'Cuora_amboinensis.fa', 'Cuora_amboinensis']
[2023-01-03T14:48:41-0800] [MainThread] [I] [toil.job] Saving graph of 1 jobs, 0 new
[2023-01-03T14:48:41-0800] [MainThread] [I] [toil.job] Processing job 'sanitize_fasta_header' kind-sanitize_fasta_header/instance-frrj58d5 v1
Traceback (most recent call last):
  File "/projects/koneill_prj/conda/envs/cactus/lib/python3.11/site-packages/toil/deferred.py", line 190, in open
    yield defer
  File "/projects/koneill_prj/conda/envs/cactus/lib/python3.11/site-packages/toil/worker.py", line 394, in workerScript
    with fileStore.open(job):
  File "/projects/koneill_prj/conda/envs/cactus/lib/python3.11/contextlib.py", line 144, in __exit__
    next(self.gen)
  File "/projects/koneill_prj/conda/envs/cactus/lib/python3.11/site-packages/toil/fileStores/nonCachingFileStore.py", line 90, in open
    os.remove(self.jobStateFile)
FileNotFoundError: [Errno 2] No such file or directory: '/var/run/user/1639/toil/7ef2404a34a458d48a9c9acb198ad067/tmpvcyunidc.jobState'
```

I've reported the error in https://github.com/ComparativeGenomicsToolkit/cactus/issues/871, but it isn't getting a response so far, and the underlying issue seems to be with Toil, at least based on the below conversation:

https://pullanswer.com/questions/toil-uses-var-run-user-uid-if-it-sees-it-whether-or-not-it-is-actually-available-to-toils-current-session

That conversation seems to imply that Toil should fall back to using another location for lock files (e.g. I have given it the location of the locally-mounted tmp space), but instead it's just failing. 

How can I force Toil to fall back to using somewhere it can actually write to?

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/TOIL-1256)
┆Issue Number: TOIL-1256

**Comment by oneillkza** [comment]
Thanks! That makes sense.

Right, I'll try and get the dev version installed -- it seems like it's not as straightforward as building the release, and is requiring me to update things like gcc and such in the virtual env.

**Comment by DailyDreaming** [comment]
@adamnovak Sounds good.

@oneillkza I'll cut a new release today.

**Comment by oneillkza** [comment]
Thanks! Apparently the issue with stuff needing to recompile was because I was using too new a version of Python (3.11). Downgrading to 3.10 allowed it to install fairly cleanly due to more precompiled packages being available?

But it will definitely be much cleaner to just use the release.

Anyway, using the dev version and setting `TOIL_COORDINATION_DIR` seems to have solved this particular issue.

**Comment by adamnovak** [comment]
Good to hear!

I'll close this as being fixed in the [5.8.0 release](https://github.com/DataBiosphere/toil/releases/tag/releases%2F5.8.0).
